### PR TITLE
Add helper class for dealing with C arrays

### PR DESCRIPTION
**Pull request by kustosz**
Manually dealing with this kind of data started to annoy me, so I guess it should have an automated flow.

### CODE REVIEW

File: stdlib/Std/src/Foreign/C/Value.luna [hunk 1]
```diff
@@ -31,6 +31,41 @@ class CString:
...
+    def make: ArrayVal
+
+    def alloc size:
+        ptr = self.ptr.mallocElems size
+        ArrayVal size ptr
+
+    def free:
+        self.ptr.free
+
+    def readAt index: self . ptr . moveElems index . read
```
Comment (mikusp) [inline stdlib/Std/src/Foreign/C/Value.luna]: Should we provide functions that check array bounds?

Comment (iamrecursion) [reply stdlib/Std/src/Foreign/C/Value.luna]: My instinct would be to default to bounds checked, as we want to provide a nicer interface to the underlying C.

Comment (mikusp) [reply stdlib/Std/src/Foreign/C/Value.luna]: I think default bounds checking is the way to go

**Comment by iamrecursion** [comment]
Can we label this?

File: stdlib/Std/src/Foreign/C/Value.luna [hunk 2]
```diff
@@ -31,6 +31,55 @@ class CString:
...
+
+    def assertInBounds index:
+        inBounds = (index < self.size) . and (index >= 0)
+        when (inBounds.not) (throw "Array index out of bounds.")
+
+    def uncheckedReadAt index:
+        self . ptr . moveElems index . read
+
+    def readAt index:
+        self.assertInBouds index
```
Comment (mikusp) [inline stdlib/Std/src/Foreign/C/Value.luna]: typo: Bouds

**Pull request merged**